[김동희] Sprint2

Heedong0924 opened the pull request:
## 요구사항

### 기본

- Github에 PR(Pull Request)을 만들어서 미션을 제출합니다.
- Netlify에 파일 배포가 아닌 포크한 Github 레포지토리로 연결합니다.
- React와 같은 UI 라이브러리를 사용하지 않고 진행합니다.

Netlify 배포 링크
https://heedong0924-pandamarket.netlify.app/

### 공통

- [x] 아래로 스크롤 해도 상단 네비게이션 바(Global Navigation Bar)가 최상단에 고정됩니다.
- [x] "판다마켓" 클릭 시 루트 페이지("/")로 이동합니다.(새로고침)
- [x] 로그인 페이지, 회원가입 페이지 모두 로고 위 상단 여백이 동일합니다.
- [x] SNS 아이콘들은 클릭시 각각 실제 서비스 홈페이지로 이동합니다.

### 로그인 페이지
- [x] "회원가입"버튼 클릭 시 "/signup" 페이지로 이동합니다.

### 회원가입 페이지
- [x] "로그인"버튼 클릭 시 "/login" 페이지로 이동합니다.

### 심화

- [x] palette에 있는 color값들을 css 변수로 등록하고 사용해 주세요.
- [x] 비밀번호 input 요소 위에 비밀번호를 확인할 수 있는 아이콘을 추가해 주세요.

## 주요 변경사항

- utility 스타일시트 삭제 (코드 리뷰 사항 반영)
- aria-label 추가 (코드 리뷰 사항 반영)
- 외부 링크 보안 추가 (코드 리뷰 사항 반영)
- 코드상 위치에 의존되었던 클래스 네이밍 변경 (코드 리뷰 사항 반영)

## 스크린샷

![image](https://github.com/user-attachments/assets/78d0729d-d5d2-4648-b0e8-d3c00c37265f)

![image](https://github.com/user-attachments/assets/7819bacb-981b-4de3-88d3-9e5c1127feb8)

## 멘토에게

- 사실 utility 클래스를 활용하게된 원인이 피그마 디자인 상으로는 컬러 팔레트나, 폰트 규격이 존재하는데 css에서 변수를 사용하는 방법이 없어서 제 나름대로 고려했던거였습니다.
- 그런데 스프린트 미션2 심화 요구사항에 css 변수 활용이 있었네요! variable.css 파일 내에 컬러 팔레트를 추가했습니다. 다만, 컬러와 다르게 폰트는 size, line-height, weight 등 사용되는 값이 많아서 이런 것들도 변수로 활용하면 좋을지 여쭤보고 싶습니다.
- 스프린트 미션1에 대한 피드백은 스프린트 미션2에서 반영하는게 방식으로 이해했는데 맞나요??
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

Heedong0924 committed:
Initial commit

Heedong0924 committed:
feat: Add a sticky gnb in index.html

Heedong0924 committed:
fix: Delete utility.css and Include code reviews

Heedong0924 committed:
fix: Fix universal selector in global.css

Heedong0924 committed:
feat: Add login.html and sign_up.html

addiescode-sj commented on img/Img_home_01.png:
투명 배경을 지원할 필요가 없으면 png보다는 jpg 파일 사용을 권장드립니다.

추천 드리자면:

- 벡터 기반의 장점을 활용할수있거나, path 정보가 간단한 로고, 아이콘등의 파일: svg
- 투명 배경을 지원할 필요가 있는 큰 이미지 파일: png
- 투명 배경을 지원할 필요가 없는 큰 이미지 파일: 최소 2x의 고배율 jpg, jpeg => 고배율 이미지를 써야하므로 png, jpeg보다 압축률이 좋은 webp로 변환해 picture태그와 함께 사용 추천

컨텐츠의 특성 혹은 몇가지 조건에 따라 올바른 파일 포맷을 고르는것도 최적화에 포함됩니다.
제가 추천드린것말고도 여러가지 참고해보시고 적용해보세요 :)

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,171 @@
+      <div class="wrap">
+        <div class="logo">
+          <a href="/" aria-label="홈으로 이동">
+            <img
+              class="img-lg"
+              src="./img/logo/lg.png"
+              alt="판다마켓 로고 이미지"
+            />
```
아마 저번 PR에서 물어보신것같은데 이미 닫힌 상태라서 제가 답변을 안드린 부분이네요:)
이 경우에는 CLS를 방지하기위해 고정 width, height를 html에서 써주면 더 좋습니다.
이유는 로고 이미지의 경우 스크린 사이즈가 커진다고하더라도 같이 커지지않고 크기를 유지하기때문입니다 :)

Heedong0924 replied:
고정 사이즈의 이미지는 CLS 방지를 위해 html에 고정 width와 height 작성

인지했습니다 감사합니다.

addiescode-sj commented on login.html:
```diff
@@ -0,0 +1,58 @@
+        </a>
+        <form method="POST" class="login-form">
+          <div class="input-box">
+            <label for="email">이메일</label>
```
label을 잘 활용해주셨네요! 굳굳 👍

addiescode-sj commented on login.html:
```diff
@@ -0,0 +1,58 @@
+        <form method="POST" class="login-form">
+          <div class="input-box">
+            <label for="email">이메일</label>
+            <input id="email" type="email" placeholder="입력" />
```
input에 name 속성이 빠져있네요!
name 속성은 서버로 폼 데이터를 전송할때 필드 이름으로 사용되는데,
이때 개별 form field 요소에 대한 식별자 역할을 하게됩니다.

따라서 서버에 폼 데이터를 혼란없이 전달하고싶다면 name 속성을 쓰시는게 필수적이겠죠?

추가적으로 autocomplete="email"을 사용하면, 브라우저가 이전에 입력한 이메일 주소를 자동으로 제안해주면서, 브라우저가 이메일 필드임을 명확히 인식할수 있어 더 정확한 자동완성을 사용자에게 제공해줄수있어요 👍

Heedong0924 replied:
빠뜨렸네요! 수정하겠습니다.

addiescode-sj commented on style/index_style.css:
```diff
@@ -0,0 +1,227 @@
+
+.banner .wrap {
+  min-width: 1080px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```
요렇게 자주 쓰게 되는 스타일 조합이 있다면 유틸리티 클래스로 만들어서 global 파일에 두면 코드 중복도 줄이고 쓰기 좋겠죠? :)

```css
/* Utility Classes */
.flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}
```

Heedong0924 replied:
저번 커밋에서 코드 리뷰 사항에 따라 utility.css 를 삭제했었는데,
결론은 개발자의 주관에 따라 (코드 가속성과 효율성을 따져가며)
조절해서 사용하라는 느낌 맞을까요??
과하면 안 좋고, 적절히 사용하면 좋고

addiescode-sj replied:
네 정확히 이해해주셨네요. 맞습니다 👍 

유틸리티 클래스 접근법으로 일괄 사용 (X) => 이유: 현재 프로젝트 구조와는 맞지 않음
다만 코드 중복 제거와 가속성을 위해 일부 사용 (O)

addiescode-sj replied:
@Heedong0924

addiescode-sj commented on style/variable.css:
변수 파일만 따로 만드셨군요! 유지보수하기가 쉽죠 👍  굳굳

css 변수로 컬러코드뿐만 아니라, 
타이포그래피, 간격, 레이아웃도 자주 사용하는값이 있으면 정의할수있어요.
z-index 값 또한 랜덤한 숫자를 넣는것보다 엘리먼트의 계층이 명확히 드러나게끔 써주는게 좋기때문에,
저는 z-index값도 구조화해서 관리하는걸 선호합니다 :)

아래 예시보시고, 참고해보시면 좋을것같네요 👍 

```css
:root {
  /* Colors */
  --color-primary: #3692ff;
  --color-primary-dark: #1967d6;
  --color-primary-darker: #1251aa;
  --color-disabled: #9ca3af;
  --color-background-light: #fcfcfc;
  --color-background-blue: #cfe5ff;
  --color-text-dark: #111827;
  --color-text-light: #e5e7eb;
  --color-text-gray: #9ca3af;
  --color-white: #ffffff;

  /* Typography */
  --font-primary: "Pretendard-Regular", sans-serif;
  --font-secondary: "ROKAF SANS", sans-serif;

  /* Spacing */
  --spacing-xs: 5px;
  --spacing-sm: 10px;
  --spacing-md: 16px;
  --spacing-lg: 24px;
  --spacing-xl: 32px;
  --spacing-xxl: 64px;

  /* Layout */
  --container-width: 1520px;
  --container-min-width: 820px;
  --content-width: 1110px;
  --feature-width: 988px;
  
  /* z-index 기본 레이어 정의 */
  --z-negative: -1;
  --z-base: 0;
  --z-low: 10;
  --z-mid: 100;
  --z-high: 1000;
  --z-top: 10000;
  
  /* 구체적인 UI 요소별 z-index 값 */
  --z-background: var(--z-negative);
  --z-default: var(--z-base);
  --z-content: var(--z-low);
  --z-dropdown: var(--z-mid);
  --z-sticky: calc(var(--z-mid) + 10);
  --z-navigation: calc(var(--z-mid) + 20);
  --z-overlay: var(--z-high);
  --z-modal: calc(var(--z-high) + 10);
  --z-notification: calc(var(--z-high) + 20);
  --z-tooltip: calc(var(--z-high) + 30);
  --z-loader: var(--z-top);
}

Heedong0924 replied:
z-index 변수는 도움이 많이 될 것 같습니다. 감사합니다!

addiescode-sj reviewed (approved):
수고하셨습니다!

전체적으로 중복제거나 구조화도 잘 신경써주시고, html, css 사용에 대한 이해도도 높으시네요 👍 
이번에도 여러가지 더 고민해보고 생각할 부분들을 피드백으로 남겨봤습니다.

## 주요 리뷰 포인트
- 올바른 이미지 포맷 적용
- CLS 방지를 위한 이미지 최적화
- input에 name 속성 사용하기
- CSS 변수

addiescode-sj commented:
## 질문에 대한 답변 

> ## 멘토에게
> * 사실 utility 클래스를 활용하게된 원인이 피그마 디자인 상으로는 컬러 팔레트나, 폰트 규격이 존재하는데 css에서 변수를 사용하는 방법이 없어서 제 나름대로 고려했던거였습니다.
> * 그런데 스프린트 미션2 심화 요구사항에 css 변수 활용이 있었네요! variable.css 파일 내에 컬러 팔레트를 추가했습니다. 다만, 컬러와 다르게 폰트는 size, line-height, weight 등 사용되는 값이 많아서 이런 것들도 변수로 활용하면 좋을지 여쭤보고 싶습니다.
> * 스프린트 미션1에 대한 피드백은 스프린트 미션2에서 반영하는게 방식으로 이해했는데 맞나요??
> * 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

네! CSS 변수 활용하실수있습니다 :) PR 본문내에 변수 활용 측면에서 어떤걸 시도할수있는지 자세히 피드백 남겨놨습니다.
스프린트 미션1에 대한 피드백은 스프린트 미션2에서 반영 => 하셔도되고, PR 아직 열려있을때 수정하시고 수정 마치면 저한테 머지 요청 주셔도 됩니다. PR이 너무 오래 열려있는것같으면 수요일쯤 일괄적으로 닫아드리니 참고해주세요 :)

Heedong0924 committed:
fix: apply reviews feedback

1. 고정 widht, height 값 추가 (CLS 방지)
2. lgoin-form 및 sign-up-form에 누락된 name속성 추가
3. utility.css 파일 추가
4. variable.css 파일에 변수 추가

Heedong0924 committed:
style: Add missing CSS layout class for banner wrap in index.html

Heedong0924 committed:
chore: Convert imgs from png to jpg

Heedong0924 commented:
리뷰 사항 반영했습니다!
